homeassistant/components/mikrotik/__init__.py

```diff
@@ -1 +1,390 @@
 """The mikrotik component."""
+import logging
+import ssl
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_HOST, CONF_PASSWORD, CONF_USERNAME, CONF_PORT,
+    CONF_SSL, CONF_METHOD, CONF_SENSORS, CONF_BINARY_SENSORS)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.util import slugify
+from homeassistant.components.device_tracker import (
+    DOMAIN as DEVICE_TRACKER)
+from .const import (CLIENT, CONNECTING, CONNECTED, IDENTITY, MEGA,
+                    CONF_ARP_PING, CONF_WAN_PORT, CONF_TRACK_DEVICES,
+                    CONF_LOGIN_METHOD, CONF_ENCODING, DEFAULT_ENCODING,
+                    MTK_DEFAULT_WAN, ARP, DHCP, WIRELESS, CAPSMAN,
+                    SENSOR, BINARY_SENSOR, MIKROTIK_SERVICES,
+                    SENSORS, BINARY_SENSORS, ATTRIB_DEVICE_TRACKER)
+
+_LOGGER = logging.getLogger(__name__)
+
+NAME = 'Mikrotik'
+DOMAIN = 'mikrotik'
+MIKROTIK = DOMAIN
+
+MTK_DEFAULT_API_PORT = '8728'
+MTK_DEFAULT_API_SSL_PORT = '8729'
+MTK_LOGIN_PLAIN = 'plain'
+MTK_LOGIN_TOKEN = 'token'
+
+MIKROTIK_SCHEMA = vol.All(
+    vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_METHOD): cv.string,
+        vol.Optional(CONF_LOGIN_METHOD):
+            vol.Any(MTK_LOGIN_PLAIN, MTK_LOGIN_TOKEN),
+        vol.Optional(CONF_PORT): cv.port,
+        vol.Optional(CONF_SSL, default=False): cv.boolean,
+        vol.Optional(CONF_ENCODING, default=DEFAULT_ENCODING): cv.string,
+        vol.Optional(CONF_TRACK_DEVICES, default=True): cv.boolean,
+        vol.Optional(CONF_ARP_PING, default=False): cv.boolean,
+        vol.Optional(CONF_SENSORS): vol.All(
+            cv.ensure_list, [vol.In(SENSORS)]),
+        vol.Optional(CONF_BINARY_SENSORS):
+            vol.All(cv.ensure_list, [vol.In(BINARY_SENSORS)]),
+        vol.Optional(CONF_WAN_PORT, default=MTK_DEFAULT_WAN): cv.string,
+    })
+)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.All(cv.ensure_list, [MIKROTIK_SCHEMA])
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass, config):
+    """Set up the Mikrotik component."""
+    hass.data[MIKROTIK] = {}
+    hass.data[CLIENT] = MikrotikAPI(hass, config[DOMAIN])
+
+    for device in config[DOMAIN]:
+        host = device[CONF_HOST]
+        hass.data[MIKROTIK][host] = {}
+        hass.data[CLIENT].connect_to_device(host)
+
+        _LOGGER.debug("[%s] Loading Mikrotik device intrgrations.", host)
+
+        sensors = device.get(CONF_SENSORS)
```

Comment:
Please remove all code related to sensor and binary sensor platforms.

Reply:
Ok, removing the sensor and binary sensor code